crates/nu-protocol/src/eval_const.rs

```diff
@@ -288,12 +291,44 @@ pub fn eval_constant(
             }
             Ok(Value::list(output, expr.span))
         }
-        Expr::Record(fields) => {
+        Expr::Record(items) => {
             let mut record = Record::new();
-            for (col, val) in fields {
-                // avoid duplicate cols.
-                let col_name = value_as_string(eval_constant(working_set, col)?, expr.span)?;
-                record.insert(col_name, eval_constant(working_set, val)?);
```

Comment:
Breaking change pin: This will turn duplication in const contexts into a hard error, which was not the case previously.